src/main/java/io/confluent/connect/hdfs/storage/HdfsStorage.java

```diff
@@ -94,7 +94,7 @@ public OutputStream create(String filename, boolean overwrite) {
   public OutputStream create(String filename, HdfsSinkConnectorConfig conf, boolean overwrite) {
     try {
       Path path = new Path(filename);
-      return path.getFileSystem(conf.getHadoopConfiguration()).create(path);
+      return FileSystem.newInstance(path.toUri(), conf.getHadoopConfiguration()).create(path);
```

Comment:
@kkonstantine, I don't see where this particular method is called in the HDFS connector, but maybe I'm missing it. Obviously we have to implement this correctly, but it would help to understand the call path to understand whether the `OutputStream` is getting closed correctly. Can you think of another way to close the `FileSystem` other than to create a delegate impl of `OutputStream` that closes the `FileSystem` instance in addition to the stream? Or can we close the `FileSystem` before the `OutputStream` is closed??

Reply:
I'm also a little confused why we can't use the `fs` property passed into the constructor here as we do in the `boolean create(...)` method.

Reply:
@rhauch, I'd rather just check equality on the config and throw an "UnsupportedOperation" in the case we don't support. I don't really want to overengineer a case that we suspect isn't even used. If we pass local and system tests, are we safe to throw Unsupported? My suspicion is that, even if it is called, it's called by line 90 which just passes "this.conf".

Reply:
You can see `append(...)` is also unsupported

Reply:
I think throwing unsupported operation is adding unnecessary risk, unless we are 100% sure this method is not called. Yes, it's additional complexity that we *might* need, but if we're not 100% sure then we have to code for that situation. Fixing issues like this is all about mitigating risk.